nengo/_vendor/npconv2d/conv2d.py

```diff
@@ -202,6 +202,39 @@ def conv2d(x, w, pad='SAME', stride=(1, 1)):
     return y
 
 
+def conv2d_groups(x, w, pad='SAME', stride=(1, 1)):
+    """2D convolution (technically speaking, correlation). Compatible with groups > 1
+    Args:
+        x: [N, H, W, C]
+        w: [I, J, C/G, K]
+        pad: [PH, PW]
+        stride: [SH, SW]
+    Returns:
+        y: [N, H', W', K]
+    """
+    ksize = w.shape[:2]
+    c = x.shape[-1]
+
+    # infer number of groups
+    assert c % w.shape[2] == 0 
```

Comment:
Add a message like `f"Number of channels ({c}) must be divisible by number of groups ({w.shape[2]})"`

Reply:
Actually, I guess `w.shape[2]` isn't the number of groups, it's the channels divided by groups. I'm not quite sure what to call it...

Reply:
Maybe "channels per group"? Thus e.g. `f"Number of channels ({c}) must be divisible by channels-per-group ({w.shape[2]})"`